packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1469,22 +1531,18 @@ export class TransactionController extends BaseController<
         delete txParams.gasPrice;
       }
 
-      const unsignedEthTx = this.prepareUnsignedEthTx(txParams);
-      this.inProcessOfSigning.add(transactionId);
-      const signedTx = await this.sign(unsignedEthTx, from);
-      await this.updateTransactionMetaRSV(transactionMeta, signedTx);
-      transactionMeta.status = TransactionStatus.signed;
-      this.updateTransaction(
-        transactionMeta,
-        'TransactionController#approveTransaction - Transaction signed',
-      );
+      const rawTx = await this.signTransaction(transactionMeta);
+
+      if (!this.beforePublish(transactionMeta)) {
+        log('Skipping publishing transaction based on hook');
+        return;
+      }
+
+      if (!rawTx) {
+        log('Skipping signed status as no signed transaction');
```

Comment:
I don't think we need any log here, as otherwise will we end up with two log messages always?